angular/lib/src/core/change_detection/differs/default_iterable_differ.dart

```diff
@@ -13,9 +13,32 @@ typedef void DefaultIterableCallback(
   int currentIndex,
 );
 
-typedef dynamic TrackByFn(int index, dynamic item);
-
-var trackByIdentity = (int index, dynamic item) => item;
+/// A function that can be used to return a unique key for [item] at [index].
+///
+/// By default, [item] itself is used as the key to track to instantiate a new
+/// template per item in `*ngFor`. If the data ever changes, AngularDart will
+/// consider any object with a different identity (`identical`) to be different
+/// - and destroy and re-create a new node.
+///
+/// To optimize performance when you have a way to determine uniqueness other
+/// than identity, such as an `id` field on an object returned from the server,
+/// you may specify a [TrackByFn]`:
+/// ```
+/// class MyComp {
+///   Object trackByEmployeeId(int index, Object item) => (item as Employee).id;
+/// }
+///
+/// class Employee {
+///   int id;
+/// }
+/// ```
+///
+/// **NOTE**: It is not safe to simply _assume_ that the second parameter is
```

Comment:
If this is worth pointing out, should we instead perform an `is` check for type promotion rather than explicitly cast?

Reply:
Right. Is something like this what you might have had in mind?

```dart
Object trackByEmployeeId(int _, Object item) => item is Employee ? item.id : item;
```

Reply:
That's another option, sure.